omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * Tests CRUD operations for {@link AutoGenerationOption}s via web service calls
+ */
+public class AutoGenerationOptionControllerTest extends MainResourceControllerTest {
+	
+	@Autowired
+	private IdentifierSourceService identifierSourceService;
+	
+	public static final String NEW_LOCATION = "9356400c-a5a2-4532-8f2b-2361b3446eb8";
+	
+	public static final String NEW_SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a902";
+	
+	public static final String LOCATION = "8d6c993e-c2cc-11de-8d13-0010c6dffd0f";
+	
+	public static final String SOURCE = "0d47284f-9e9b-4a81-a88b-8bb42bc0a901";
+	
+	public static final String IDENTIFIER_TYPE_UUID = "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c";
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+	}
+	
+	@Override
+	public String getUuid() {
+		return IDENTIFIER_TYPE_UUID;
+	}
+	
+	@Override
+	public long getAllCount() {
+		Integer allAutoGenerationOptions = 0;
+		for (PatientIdentifierType patientIdentifierType : Context.getPatientService().getAllPatientIdentifierTypes()) {
+			allAutoGenerationOptions += identifierSourceService.getAutoGenerationOptions(patientIdentifierType).size();
+		}
+		return allAutoGenerationOptions;
+	}
+	
+	@Test
+	public void shouldGetAnAutoGenerationOptionByPatientIdentifierType() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		PatientIdentifierType patientIdentifier = Context.getPatientService().getPatientIdentifierTypeByUuid(getUuid());
+		AutoGenerationOption autoGenerationOption = identifierSourceService.getAutoGenerationOption(patientIdentifier);
+		assertEquals(autoGenerationOption.getIdentifierType().getUuid(),
+		    PropertyUtils.getProperty(result, "identifierType"));
+		assertEquals(autoGenerationOption.getLocation().getUuid(), PropertyUtils.getProperty(result, "location"));
+		assertEquals(autoGenerationOption.getSource().getUuid(), PropertyUtils.getProperty(result, "source"));
+	}
+	
+	@Test
+	public void shouldCreateNewAutoGenerationOption() throws Exception {
+		long originalCount = getAllCount();
+		boolean automaticGenerationEnabled = true;
+		boolean manualEntryEnabled = true;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", getUuid());
+		autoGenerationOption.add("location", LOCATION);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
```

Comment:
Can you add assertions to confirm that it was created with all the above properties?

Reply:
done below in line 92 onwards